src/legacy/core_plugins/kibana/public/discover/np_ready/angular/helpers/_get_aspects.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Table, Dimensions, Aspects } from './point_series';
+
+/**
+ * Identify and group the columns based on the aspect of the pointSeries
+ * they represent.
+ *
+ * @return {object} - an object with a key for each aspect (see map). The values
+ *                    may be undefined, a single aspect, or an array of aspects.
+ */
+export function getAspects(table: Table, dimensions: Dimensions) {
+  const aspects: Aspects = {} as Aspects;
+  (Object.keys(dimensions) as Array<keyof Dimensions>).forEach(name => {
```

Comment:
Since we know Discover's[ `dimensions` object ](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js#L862)exactly, I simplified all functions for Discover to get rid of unnecessary code.